Sources/ArcGISToolkit/Components/OfflineMapAreasView/OfflineMapAreasView.swift

```diff
@@ -72,7 +83,10 @@ public struct OfflineMapAreasView: View {
         case .success(let models):
             if mapViewModel.hasPreplannedMapAreas {
                 List(models) { preplannedMapModel in
-                    PreplannedListItemView(model: preplannedMapModel)
+                    PreplannedListItemView(
+                        mapViewModel: mapViewModel,
```

Comment:
Currently, there are 2 usages of the `mapViewModel` in the `PreplannedListItemView`:

- `mapViewModel.canShowNotifications`
- `mapViewModel.setPreplannedMobileMapPackages()`

For the first Boolean, I think it might be OK to keep it on the model; it can also be passed in as a initializer parameter.

For the second method, I feel it should be a separate different method in the `preplannedMapModel`. Because in the `MapViewModel`, the method is used to set a series of mobile map packages from downloaded local files. Whereas in the `PreplannedListItemView`, each download button corresponds to a single mobile map package, so it doesn't need to scan through all downloaded packages to determine which mobile map package is currently being held.

If we can have a separate method, then I suggest not passing the `mapViewModel` to  `PreplannedListItemView` to reduce complexity.

Reply:
Good point, usage of the `mapViewModel` has been removed from `PreplannedListItemView`.